crates/frontend/src/pages/custom_types/mod.rs

```diff
@@ -0,0 +1,234 @@
+                        let row_data = serde_json::from_value::<CustomTypeRow>(
+                            delete_row_json.clone(),
+                        )
+                        .unwrap();
+                        spawn_local({
+                            async move {
+                                let tenant = tenant_rs.get();
+                                let _ =
+                                    delete_type(tenant, row_data.clone().type_name).await;
+                                types_resource.refetch();
+                            }
+                        });
+                    };
+                    view! {
+                        <div class="join">
+                            <span class="cursor-pointer" on:click=edit_click>
+                                <i class="ri-pencil-line ri-xl text-blue-500"></i>
+                            </span>
+                            <span class="cursor-pointer" on:click=delete_click>
+                                <i class="ri-delete-bin-line ri-xl text-red-500"></i>
+                            </span>
+                        </div>
+                    }
+                    .into_view()
+                },
+            ),
+        ]
+    });
+
+    view! {
+        {move || {
+            let handle_close = move || {
+                close_drawer(TYPE_DRAWER_ID);
+                selected_type.set(None);
+            };
+            if let Some(selected_type_data) = selected_type.get() {
+                view! {
+                    <Drawer
+                        id=TYPE_DRAWER_ID.to_string()
+                        header="Edit Type Template"
+                        handle_close=handle_close
+                    >
+                        <TypeTemplateForm
+                            type_name=selected_type_data.type_name
+                            type_schema=selected_type_data.type_schema
+                            handle_submit=move || {
+                                types_resource.refetch();
+                                selected_type.set(None);
+                                close_drawer(TYPE_DRAWER_ID);
+                            }
+                        />
+                    </Drawer>
+                }
+            } else {
+                view! {
+                    <Drawer
+                        id=TYPE_DRAWER_ID.to_string()
+                        header="Create New Type Template"
+                        handle_close=handle_close
+                    >
+                        <TypeTemplateForm handle_submit=move || {
+                            types_resource.refetch();
+                            selected_type.set(None);
+                            close_drawer(TYPE_DRAWER_ID);
+                        }/>
+
+                    </Drawer>
+                }
+            }
+        }}
+
+        <div class="p-8">
+            <Suspense fallback=move || view! { <p>"Loading..."</p> }>
```

Comment:
We have a `Skeleton` component now, that can be used here, instead of `Loading` text

Reply:
Got it, will add it here